### PR TITLE
feat: brand new comments with replies

### PR DESCRIPTION
**Pull request by JayeshVP24**
## Screenshots (if appropriate)

### CODE REVIEW

File: src/config/chakra.config.ts [hunk 1]
```diff
@@ -28,6 +29,9 @@ const theme = extendTheme({
       },
     },
   },
+  fontSizes: {
+    xxs: '0.625rem',
```
Comment (a20hek) [inline src/config/chakra.config.ts:33]: we already have xxs – `xx-small`

File: src/features/listings/components/ListingPage/Comments.tsx [hunk 2]
```diff
@@ -8,25 +9,26 @@ import {
...
 import { LoginWrapper } from '@/components/LoginWrapper';
 import { ErrorInfo } from '@/components/shared/ErrorInfo';
 import { Loading } from '@/components/shared/Loading';
 import { UserAvatar } from '@/components/shared/UserAvatar';
 import type { Comment } from '@/interface/comments';
 import { userStore } from '@/store/user';
-import { dayjs } from '@/utils/dayjs';
-import { getURL } from '@/utils/validUrl';
 
+import { autoResize } from '../../utils';
```
Comment (a20hek) [inline src/features/listings/components/ListingPage/Comments.tsx:22]: we already have an autoresize component: 
https://github.com/SuperteamDAO/earn/blob/main/src/components/shared/autosize-textarea.tsx

**Comment by JayeshVP24** [comment]
@JayeshVP24

File: src/features/listings/components/ListingPage/Comments.tsx [hunk 3]
```diff
@@ -84,6 +127,28 @@ export const Comments = ({ refId, refType }: Props) => {
     }
   };
 
+  useEffect(() => {
```
Comment (a20hek) [inline src/features/listings/components/ListingPage/Comments.tsx:130]: what's going on here?

Comment (JayeshVP24) [reply src/features/listings/components/ListingPage/Comments.tsx:130]: whenever use makes a submission, we want a comment in their name automatically
comment goes through but doesn't reflect instantly when scrolled down to the comments.

i have a custom event dispatcher where user submission is done, when the event `update-comments` is dispatched, we listen here and getComments again.

better this way, no downside, do not have to send all comments logic back to page and perform a huge prop drilling.

File: src/features/listings/components/Submission/SubmissionModal.tsx [hunk 4]
```diff
@@ -187,6 +190,16 @@ export const SubmissionModal = ({
...
       });
 
+      if (!editMode) {
+        await axios.post(`/api/comment/create`, {
+          message: randomSubmissionCommentGenerator(type),
+          listingId: id,
+          submissionId: response?.data?.id,
+          type: 'SUBMISSION',
+        });
+        window.dispatchEvent(new Event('update-comments'));
```
Comment (JayeshVP24) [inline src/features/listings/components/Submission/SubmissionModal.tsx:200]: @a20hek 
check this

**Review by a20hek — CHANGES_REQUESTED** [review]
image

the width of the two containers are not consistent

**Comment by JayeshVP24** [comment]
> image
> the width of the two containers are not consistent

Solved

**Pull request merged**